src/graph/expression_graph.h

```diff
@@ -232,6 +232,18 @@ class ExpressionGraph : public std::enable_shared_from_this<ExpressionGraph> {
     namespace_ = newNamespace;
   }
 
+  const std::unordered_map<std::string, Expr> & getParamsNamedMap() const {
```

Comment:
It seems that this isn't called by anything

Reply:
Correct, we should kill it.  @XapaJIaMnu

Reply:
Oh, now you're doing your own parameter indexing to avoid `std::string` matches?

Reply:
Yeah guess I can remove this code given that I removed the need to use it.  And with that, do we really need to patch Marian at all?

Reply:
No.... They have a CudaMemCPY but it's not public.. We can do that so that we avoid using custom CUDA code... Also we should purge std::map wherever possible